ads/google/a4a/test/test-traffic-experiments.js

```diff
@@ -77,186 +70,71 @@ describe('all-traffic-experiments-tests', () => {
-          'experiment is on').to.be.true;
-      expect(getPageExperimentBranch(sandbox.win, 'testExperimentId')).to.equal(
-          testExperimentSet['testExperimentId'].control);
-    });
-    it('handles experiment diverted path: experiment', () => {
-      // Force experiment on.
-      toggleExperiment(sandbox.win, 'testExperimentId', true, true);
-      // Force the experiment branch to be chosen by making the accurate PRNG
-      // return a value > 0.5.
-      RANDOM_NUMBER_GENERATORS.accuratePrng.onFirstCall().returns(0.6);
-      randomlySelectUnsetPageExperiments(sandbox.win, testExperimentSet);
-      expect(isExperimentOn(sandbox.win, 'testExperimentId'),
-          'experiment is on').to.be.true;
-      expect(getPageExperimentBranch(sandbox.win, 'testExperimentId')).to.equal(
-          testExperimentSet['testExperimentId'].experiment);
-    });
-    it('handles multiple experiments', () => {
-      toggleExperiment(sandbox.win, 'expt_0', true, true);
-      toggleExperiment(sandbox.win, 'expt_1', false, true);
-      toggleExperiment(sandbox.win, 'expt_2', true, true);
-      toggleExperiment(sandbox.win, 'expt_3', true, true);
-
-      const experimentInfo = {
-        'expt_0': {
-          control: '0_c',
-          experiment: '0_e',
-        },
-        'expt_1': {
-          control: '1_c',
-          experiment: '1_e',
-        },
-        'expt_2': {
-          control: '2_c',
-          experiment: '2_e',
-        },
-        // expt_3 omitted.
-      };
-      RANDOM_NUMBER_GENERATORS.accuratePrng.returns(0.6);
-      randomlySelectUnsetPageExperiments(sandbox.win, experimentInfo);
-      expect(isExperimentOn(sandbox.win, 'expt_0'),
-          'expt_0 is on').to.be.true;
-      expect(isExperimentOn(sandbox.win, 'expt_1'),
-          'expt_1 is on').to.be.false;
-      expect(isExperimentOn(sandbox.win, 'expt_2'),
-          'expt_2 is on').to.be.true;
-      // Note: calling isExperimentOn('expt_3') would actually evaluate the
-      // frequency for expt_3, possibly enabling it.  Since we wanted it to be
-      // omitted altogether, we'll evaluate it only via its branch.
-      expect(getPageExperimentBranch(sandbox.win, 'expt_0')).to.equal(
-          '0_e');
-      expect(getPageExperimentBranch(sandbox.win, 'expt_1')).to.not.be.ok;
-      expect(getPageExperimentBranch(sandbox.win, 'expt_2')).to.equal(
-          '2_e');
-      expect(getPageExperimentBranch(sandbox.win, 'expt_3')).to.not.be.ok;
+    it('should enable the external A4A experiment from the URL', () => {
+      const externalBranches = {control: '12', experiment: '34'};
+      const internalBranches = {control: '56', experiment: '78'};
+
+      sandbox.win.location.search = '?exp=a4a:2';
+
+      const renderViaA4a = googleAdsIsA4AEnabled(
+          sandbox.win, element, 'exp_name', externalBranches, internalBranches);
+      expect(renderViaA4a).to.be.true;
+      expect(element.getAttribute(EXPERIMENT_ATTRIBUTE)).to.equal('34,2088461');
```

Comment:
Better to use `traffic-experiments.js#IsExperimentOn` to check for eid `34`, rather than literal string match -- the other active experiments could change outside of your test control.  And use `expect(IsExperimentOn(12)).to.be.false` to ensure that unwanted experiments are not turned on.  Finally, import `traffic-experiments.js#EXTERNALLY_SELECTED_ID` and check for it via `IsExperimentOn` rather than checking `2088461` directly.  Might be simplest to roll all this into a helper function that does all these expectations and can be called from every test case. (Ditto for other eid tests below.)